### PR TITLE
Cleanup Message Generation

### LINKED ISSUE #732: Message generation could break unexpectedly with weird paths being processed
The way paths are handled in `rosidl_parser/rosidl_parser.js` can be problematic if, for example, spaces are present within them. Also, errors being thrown aren't very verbose and helpful in narrowing down actual problems.

This is related to #711, and wasn't entirely fixed by #712.

### PR DESCRIPTION
**Pull request by felixdivo**
This PR closes #732 improves upon #712 by handling other whitespace besides normal spaces in a more robust way. Also cleans up parts of the message/etc. generation process (using `async function`s).

### CODE REVIEW

**Comment by felixdivo** [comment]
Don't really know if the Travis/AppVeyor failures have anything to do this this PR but it not I'm done with this set of changes and ready for a review.

**Comment by felixdivo** [comment]
I hope it's okay that I've added my name to the list of contributors.

File: README.md [hunk 1]
```diff
@@ -249,6 +249,7 @@ Special thanks to the people who contribute.
 - [Alex Mikhalev](https://github.com/amikhalev)
 - [Wayne Parrott](https://github.com/wayneparrott)
 - [Matt Richard](https://github.com/mattrichard)
+- [Felix Divo](https://github.com/felixdivo)
```
Comment (minggangw) [inline README.md:252]: Of course,  you should have added it when submitting your first patch, I forgot to remind you :)

**Review by minggangw — COMMENTED** [review]
Thanks for cleaning up the code! LGTM

**Review by minggangw — APPROVED** [review]

**Comment by minggangw** [comment]
Hi @felixdivo I noticed that the appveyor always has an error with this PR when generating JavaScript message, please check https://ci.appveyor.com/project/minggangw/rclnodejs/builds/36416972#L438
```bash
> rclnodejs@0.16.0 postinstall C:\proj\rclnodejs
> node scripts/generate_messages.js
Start JavaScript message generation...
Caught error: Error
```
Would you please take a look?

**Comment by felixdivo** [comment]
This is now fixed.

**Pull request merged**